src/utils/fetches.js

```diff
@@ -1,18 +1,71 @@
+import { getToken } from './storage';
+
 export const fetchPost = (api, obj) => {
+  const token = getToken();
+
   return fetch(api, {
     method: 'POST',
     headers: {
       'Content-Type': 'application/json',
+      Authorization: token,
     },
     body: JSON.stringify(obj),
-  }).then((response) => response.json());
+  });
 };
 
-export const fetchGet = (api, token) => {
+export const fetchDelete = (api) => {
+  // api/:id
+  const token = getToken();
+
   return fetch(api, {
-    method: 'GET',
+    method: 'DELETE',
+    headers: {
+      Authorization: token,
+    },
+  });
+};
+
+export const fetchGet = (api) => {
+  const token = getToken();
+
+  token
+    ? fetch(api, {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: token,
+        },
+      }).then((response) => response.json())
+    : fetch(api, {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }).then((response) => response.json());
```

Comment:
```suggestion
const headers = Object.assign(
  { 'Content-Type': 'application/json' },
  token ? { Authorization: token } : null
)

const options = { method: 'GET', headers }

fetch(api, options).then((response) => response.json())
```
Object.assign() 문법을 사용하면 조건부로 객체의 프로퍼티를 추가할 수 있습니다. 현재 구조로는 각 경우의 어떤 부분이 바뀌었나 바로 확인하기 힘들어보입니다!

Reply:
공용 으로 넣기로 합의하여 수정하였습니다